### PR TITLE
Feature/initial queries fixes

### PR DESCRIPTION
**Pull request by achen2401**
address this story:  https://www.pivotaltracker.com/story/show/151658990
- delay collection of incomplete fields information to allow rendering of relevant elements
- handle situations when user does not click on "continue to truenth" button but completed all fields (i.e. should be re-directed to appropriated site)

### CODE REVIEW

**Comment by mcjustin** [comment]
Seems like @pbugni might be a better reviewer than me...

File: portal/templates/initial_queries.html [hunk 1]
```diff
@@ -497,14 +497,19 @@ <h4 class="profile-item-title"> {{ _("Thank you.") }}</h4>
...
 };
+
 $(document).ready(function(){
 	/*
 	 * the flow here : 
 	 * get still needed core data
 	 * populate terms if it is agreed
-	 * get incomplete fields thereafter
+	 * get incomplete fields thereafter 
+   * note: need to delay gathering incomplete fields to allow fields to be render (e.g. broken in firefox if no delay)
```
Comment (pbugni) [inline portal/templates/initial_queries.html]: mind lining up this comment?

File: portal/views/portal.py [hunk 2]
```diff
@@ -40,10 +40,12 @@
 from ..models.reporting import get_reporting_stats
 from ..models.role import Role, ROLE, ALL_BUT_WRITE_ONLY
 from ..models.user import current_user, get_user, User, UserRoles
+from ..models.user_consent import UserConsent
 from ..system_uri import SHORTCUT_ALIAS
 from ..trace import establish_trace, dump_trace
 
 
+
```
Comment (pbugni) [inline portal/views/portal.py]: you need 2 lines between the imports and code - not 3 :)

File: portal/views/portal.py [hunk 3]
```diff
@@ -489,21 +491,36 @@ def challenge_identity(user_id=None, next_url=None, merging_accounts=False):
...
+        return next_after_login()
+    elif len(Coredata().still_needed(user)) == 0:
+        # also handle the situations that resulted from: 1. user refreshing the browser or
+        # 2. exiting browser and resuming session thereafter
+        # In both cases, the request method is GET, 
+        # hence a redirect back to initial-queries page won't ever reach the above check 
+        # specifically for next_after_login based on the request method of POST
+        # NB: user needs to have provided consent before we can check for next_after_login
+        # case in point:
+        # user has pre-selected clinic via access code BUT has not provided consent yet
```
Comment (pbugni) [inline portal/views/portal.py]: I get the need for the `elif` - but not the hand selected lookup for `UserConsent`.  There may well be a roll or product where that doesn't make sense.

If `len(Coredata().still_needed(user)) == 0` - you can safely assume we have everything needed, right?

**Comment by achen2401** [comment]
@pbugni 
Paul, I think I address your all of comments.  Let me know if you see any issue.

**Pull request merged**